geodiff/src/drivers/sqlitedriver.cpp

```diff
@@ -71,13 +83,18 @@ class Sqlite3SavepointTransaction
       assert( mDb );
       // there were no errors - release the savepoint and our changes get saved
       if ( sqlite3_exec( mDb.get()->get(), "RELEASE changeset_apply", 0, 0, 0 ) != SQLITE_OK )
+      {
+        std::string errMsg = sqliteErrorMessage( mDb.get()->get() );
+        mContext->logger().error( errMsg );
         throw GeoDiffException( "Failed to release savepoint" );
+      }
```

Comment:
if you throw, you usually want to log it upstream, but you want to keep the error msg at hand? we may want to create new exception type for sqlite3 errors? or add errMsh to exception ttext?

Reply:
This message did not contain any additional information, only this text. As far as I can see this is true for the most of the exceptions in geodiff: only user-friendly text is passed. I will change all of them then.

Reply:
the idea is that the exception iis used as error (message), so yuo do not need to pass context to places where exception is triggered.

Reply:
Ah, got it. I though we want to have more detailed info in the log. Will update messages and drop context then.

Reply:
GeoDiffException is always catched in the geodiff.cpp at least. so we can save the message to log there.

Reply:
bear in mind that log should be quite verbose: so usually you want too at least know in this function in code it happened